third_party/move/move-compiler-v2/src/pipeline/control_flow_graph_simplifier.rs

```diff
@@ -207,8 +218,12 @@ struct ControlFlowGraphCodeGenerator {
     /// Entry/exit block is mapped to an empty vector.
     code_blocks: BTreeMap<BlockId, Vec<Bytecode>>,
     /// Maps a basic block to its successors.
+    /// If a block has no successors, it is mapped to an empty vector,
+    /// and the key for that block still exists.
     successors: BTreeMap<BlockId, Vec<BlockId>>,
```

Comment:
Is it okay to have duplicate successors? i.e., `A` -> [`B`, `B`]? It might make sense from a correctness perspective to use a `BTreeSet` here instead.

Reply:
This is inherited `StacklessControlFlowGraph`, which uses vector to order the successors in the order of true/false branch (or the reverse, I can't remember, and not explicitly documented). I added another invariant for this, and this is guaranteed by `eliminate_branch_to_same_target`.